sys/net/routing/ng_rpl/ng_rpl_control_messages.c

```diff
@@ -0,0 +1,582 @@
+                ng_rpl_opt_target_t *target = (ng_rpl_opt_target_t *) opt;
+                if (first_target == NULL) {
+                    first_target = target;
+                }
+
+                kernel_pid_t if_id = ng_ipv6_netif_find_by_prefix(NULL, &dodag->dodag_id);
+                if (if_id == KERNEL_PID_UNDEF) {
+                    DEBUG("RPL: no interface found for the configured DODAG id\n");
+                    break;
+                }
+
+                fib_add_entry(if_id, target->target.u8, sizeof(ng_ipv6_addr_t), AF_INET6, src->u8,
+                        sizeof(ng_ipv6_addr_t), AF_INET6,
+                        (dodag->default_lifetime * dodag->lifetime_unit) * SEC_IN_MS);
+                break;
+            }
+            case (NG_RPL_OPT_TRANSIT): {
+                DEBUG("RPL: RPL TRANSIT INFO DAO option parsed\n");
+                ng_rpl_opt_transit_t *transit = (ng_rpl_opt_transit_t *) opt;
+                if (first_target == NULL) {
+                    DEBUG("RPL: Encountered a RPL TRANSIT DAO option without \
+a preceding RPL TARGET DAO option\n");
+                    break;
+                }
+
+                do {
+                    fib_update_entry(first_target->target.u8, sizeof(ng_ipv6_addr_t),
+                            src->u8, sizeof(ng_ipv6_addr_t), AF_INET6,
+                            (transit->path_lifetime * dodag->lifetime_unit * SEC_IN_MS));
+                    first_target = (ng_rpl_opt_target_t *) (((uint8_t *) (first_target)) +
+                        sizeof(ng_rpl_opt_t) + first_target->length);
+                }
+                while (first_target->type == NG_RPL_OPT_TARGET);
+
+                first_target = NULL;
+                break;
+            }
+        }
+        l += opt->length + sizeof(ng_rpl_opt_t);
+        opt = (ng_rpl_opt_t *) (((uint8_t *) (opt + 1)) + opt->length);
+    }
+    return;
+}
+
+void ng_rpl_recv_DIO(ng_rpl_dio_t *dio, ng_ipv6_addr_t *src, uint16_t len)
+{
+    ng_rpl_instance_t *inst = NULL;
+    ng_rpl_dodag_t *dodag = NULL;
+
+    if (ng_rpl_instance_add(dio->instance_id, &inst)) {
+        inst->mop = (dio->g_mop_prf >> NG_RPL_MOP_SHIFT) & NG_RPL_SHIFTED_MOP_MASK;
+        inst->of = (ng_rpl_of_t *) ng_rpl_get_of_for_ocp(NG_RPL_DEFAULT_OCP);
+    }
+    else if (inst == NULL) {
+        DEBUG("RPL: Could not allocate a new instance.\n");
+        return;
+    }
+
+    if ((byteorder_ntohs(dio->rank) == NG_RPL_INFINITE_RANK) &&
+            (ng_rpl_dodag_get(inst, &dio->dodag_id) == NULL)) {
+        DEBUG("RPL: ignore INFINITE_RANK DIO when we are not part of this DODAG\n");
+        ng_rpl_instance_remove(inst);
+        return;
+    }
+
+    if (ng_rpl_dodag_add(inst, &dio->dodag_id, &dodag)) {
+        DEBUG("RPL: Joined DODAG (%s).\n",
+                ng_ipv6_addr_to_str(addr_str, &dio->dodag_id, sizeof(addr_str)));
+
+        uint8_t tmp_len = len - (sizeof(ng_rpl_dio_t) + sizeof(ng_icmpv6_hdr_t));
```

Comment:
how about add 
`len -= (sizeof(ng_rpl_dio_t) + sizeof(ng_icmpv6_hdr_t));` 
at the beginning of this function and remove this line and l:366 ?